### PR TITLE
MCO-392: Start using `rhel-coreos` image rather than `machine-os-content`

### PR DESCRIPTION
**Pull request by jlebon**
The `machine-os-content` is deprecated and we'd like to stop shipping it entirely in the release payload. Point the DTK imagestream to the `rhel-coreos` image instead.

Closes: #101

### CODE REVIEW

**Comment by jlebon** [comment]
This would go at the same time as https://github.com/openshift/os/pull/1374.
We need to make sure DTK, ART, and RHCOS teams are aligned before doing this change.

**Comment by ybettan** [comment]
Thanks for this PR @jlebon.

Is https://github.com/openshift/os/pull/1374 adding the "io.openshift.build.versions": "machine-os=<ocp version>.<rhel version>.<timestamp>" label to the `rhel-coreos` container?

Do we need anything else to make it propagate into the payload or it is happening automatically.

If that's all we need I am good with merging.

**Review by travier — APPROVED** [review]

**Comment by LorbusChris** [comment]
This will require stopping driver-toolkit image mirroring first, as the release controller will otherwise barf when there's an unknown image (rhel-coreos does not exist in OKD imagestreams): https://github.com/openshift/release/pull/43816

**Comment by LorbusChris** [comment]
Also note that the `machine-os-content` ref is also still used by the installer: https://github.com/openshift/installer/blob/c723ac80b519bb33b8e2c9c46883c9d5d7284857/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L46

**Comment by jlebon** [comment]
> Also note that the `machine-os-content` ref is also still used by the installer: [openshift/installer@`c723ac8`/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L46](https://github.com/openshift/installer/blob/c723ac80b519bb33b8e2c9c46883c9d5d7284857/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L46)

Opened https://github.com/openshift/installer/pull/7636!

**Comment by ybettan** [comment]
@LorbusChris Do we need anything else before undrafting this PR and merging it?

**Comment by ybettan** [comment]
Great.
Please undraft the PR so we make sure CI passes.

/hold
In order to sync the merge with other PRs required.

**Comment by jlebon** [comment]
OK as expected, CI is failing with:

```
 error: failed to push image registry.build05.ci.openshift.org/ci-op-1p5hwli8/release:latest: unable to upload new layer (0): Patch "https://registry.build05.ci.openshift.org/v2/ci-op-1p5hwli8/release/blobs/uploads/c58b18af-367c-4f7e-9749-17e89f6f1635?_state=Hj3H_lI5ysPOLpl-oZTVM9_QqDZM8oTjQv0x9aPIvLl7Ik5hbWUiOiJjaS1vcC0xcDVod2xpOC9yZWxlYXNlIiwiVVVJRCI6ImM1OGIxOGFmLTM2N2MtNGY3ZS05NzQ5LTE3ZTg5ZjZmMTYzNSIsIk9mZnNldCI6MCwiU3RhcnRlZEF0IjoiMjAyMy0xMS0wOFQxNDoyOToxMy42NDcxMDI0ODZaIn0%3D": unknown version reference "machine-os" 
```

So I think we're ready here to merge them all together!

https://github.com/openshift/os/pull/1374 is ready to merge, so if you agree and override the CI failures to get this in, I'll get that one in too.

**Comment by jlebon** [comment]
Actually, we went ahead and merged https://github.com/openshift/os/pull/1374. This should be good to go too now!

**Comment by sosiouxme** [comment]
ART bits are merged too

**Comment by jlebon** [comment]
Will we actually be able to achieve green CI here or do we have to force merge it?

I do see we have an RHCOS build now with https://github.com/openshift/os/pull/1374 (415.92.202311082221-0). It was pushed as quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:12ff620fd1eebe68c006cbe0b971ed570a0d304c5992048b898c6be176dad6eb. But I'm not familiar enough with OpenShift CI internals to know if that'll end up in the right place on its own for CI to pick it up here or if e.g. it'll fail to get there anyway until this PR is forced in.

**Comment by jlebon** [comment]
> I do see we have an RHCOS build now with [openshift/os#1374](https://github.com/openshift/os/pull/1374) (415.92.202311082221-0). It was pushed as quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:12ff620fd1eebe68c006cbe0b971ed570a0d304c5992048b898c6be176dad6eb.

BTW, I can confirm that image has the expected labels:

```
$ skopeo inspect -n docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:12ff620fd1eebe68c006cbe0b971ed570a0d304c5992048b898c6be176dad6eb  | grep io.openshift.build
        "io.openshift.build.version-display-names": "machine-os=Red Hat Enterprise Linux CoreOS",
        "io.openshift.build.versions": "machine-os=415.92.202311082221-0",
```

**Comment by sosiouxme** [comment]
/retest
after removing machine-os-content

**Comment by ybettan** [comment]
/override ci/prow/e2e-aws
/override ci/prow/e2e-aws-driver-toolkit-presubmi
/override ci/prow/e2e-upgrade
/override ci/prow/images

https://redhat-internal.slack.com/archives/CB95J6R4N/p1699558337383699?thread_ts=1699386359.814909&cid=CB95J6R4N

**Pull request merged**
